rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/EditNodeSourceWindow.java

```diff
@@ -108,9 +120,116 @@ protected void populateFormValues(Label windowLabel, DynamicForm windowForm, Tex
+    }
+
+    protected void fetchNodeSourceConfigurationWithCallback(Label windowLabel, DynamicForm windowForm,
+            TextItem nodeSourceNameItem, CheckboxItem nodesRecoverableItem) {
+
+        this.controller.fetchNodeSourceConfiguration(this.nodeSourceName, () -> {
+
+            NodeSourceConfiguration nodeSourceConfiguration = this.controller.getModel()
+                                                                             .getEditedNodeSourceConfiguration();
+
+            nodeSourceNameItem.setDefaultValue(nodeSourceConfiguration.getNodeSourceName());
+
+            nodesRecoverableItem.setValue(nodeSourceConfiguration.getNodesRecoverable());
+
+            manageNodeSourceWindowItems(nodeSourceNameItem, nodesRecoverableItem);
+
+            LinkedHashMap<String, String> selectItemValues = new LinkedHashMap<>();
+
+            this.allFormItems = prepareFormItems();
+
+            PluginDescriptor focusedInfrastructurePlugin = nodeSourceConfiguration.getInfrastructurePluginDescriptor();
+            this.focusedInfrastructurePluginName = focusedInfrastructurePlugin.getPluginName();
+            this.allFormItems.add(this.infrastructureSelectItem);
+            addFocusedPluginValues(selectItemValues, focusedInfrastructurePlugin);
+            handleAdditionalInfrastructureFormItems(selectItemValues, focusedInfrastructurePlugin);
+            this.infrastructureSelectItem.setValueMap(selectItemValues);
+            this.infrastructureSelectItem.setDefaultToFirstOption(true);
+            this.previousSelectedInfrastructure = focusedInfrastructurePlugin.getPluginName();
+
+            this.allFormItems.add(new SpacerItem());
+            selectItemValues.clear();
+
+            PluginDescriptor focusedPolicyPlugin = nodeSourceConfiguration.getPolicyPluginDescriptor();
+            this.focusedPolicyPluginName = focusedPolicyPlugin.getPluginName();
+            this.allFormItems.add(this.policySelectItem);
+            addFocusedPluginValues(selectItemValues, focusedPolicyPlugin);
+            handleAdditionalPolicyFormItems(selectItemValues, focusedPolicyPlugin);
+            this.policySelectItem.setValueMap(selectItemValues);
+            this.policySelectItem.setDefaultToFirstOption(true);
+            this.previousSelectedPolicy = focusedPolicyPlugin.getPluginName();
+
+            this.infrastructureSelectItem.addChangedHandler(changedEvent -> resetFormForInfrastructureSelectChange());
+            this.policySelectItem.addChangedHandler(changedEvent -> resetFormForPolicySelectChange());
+
+            this.allFormItems = modifyFormItemsAfterCreation(focusedInfrastructurePlugin, focusedPolicyPlugin);
 
-        return formItems;
+            windowForm.setFields(this.allFormItems.toArray(new FormItem[this.allFormItems.size()]));
+            windowLabel.hide();
+            windowForm.show();
+
+            //this.allFormItems = modifyFormItemsAfterCreation(focusedInfrastructurePlugin, focusedPolicyPlugin);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)